tests/components/drop_connect/test_switch.py

```diff
@@ -0,0 +1,275 @@
+"""Test DROP switch entities."""
+
+from homeassistant.components.drop_connect.const import DOMAIN
+from homeassistant.components.switch import (
+    DOMAIN as SWITCH_DOMAIN,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+)
+from homeassistant.const import ATTR_ENTITY_ID, STATE_OFF, STATE_ON, STATE_UNKNOWN
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from .common import (
+    TEST_DATA_FILTER,
+    TEST_DATA_FILTER_RESET,
+    TEST_DATA_FILTER_TOPIC,
+    TEST_DATA_HUB,
+    TEST_DATA_HUB_RESET,
+    TEST_DATA_HUB_TOPIC,
+    TEST_DATA_PROTECTION_VALVE,
+    TEST_DATA_PROTECTION_VALVE_RESET,
+    TEST_DATA_PROTECTION_VALVE_TOPIC,
+    TEST_DATA_SOFTENER,
+    TEST_DATA_SOFTENER_RESET,
+    TEST_DATA_SOFTENER_TOPIC,
+)
+
+from tests.common import async_fire_mqtt_message
+from tests.typing import MqttMockHAClient
+
+
+async def test_switches_hub(
+    hass: HomeAssistant, config_entry_hub, mqtt_mock: MqttMockHAClient
+) -> None:
+    """Test DROP switches for hubs."""
+    config_entry_hub.add_to_hass(hass)
+    assert await async_setup_component(hass, DOMAIN, {})
+    await hass.async_block_till_done()
+
+    water_supply_switch_name = "switch.hub_drop_1_c0ffee_water_supply"
+    hass.states.async_set(water_supply_switch_name, STATE_UNKNOWN)
```

Comment:
Why are we setting a state? That shouldn't be needed.

Reply:
Missed that, only asserting the state should be enough